cpp/log/cert.cc

```diff
@@ -883,7 +914,8 @@ Cert::Status Cert::IsValidNameConstrainedIntermediateCa() const {
 
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
-  if (HasBasicConstraintCATrue() != Cert::TRUE
+  StatusOr<bool> has_ca_constraint_status = HasBasicConstraintCATrue();
+  if (StatusOrBoolToCertStatus(has_ca_constraint_status) != Cert::TRUE
```

Comment:
You're using your temporary helper just to avoid doing `(!has_ca_constraint_status.ok() || !has_ca_constraint_status.ValueOrDie())` here, but we want to delete that temporary helper, so might as well bite the bullet?